File: cppSource/cppSource/Chromosome.cpp
#include "Chromosome.h"
#include "initHelp.h"
Chromosome::Chromosome(ChromosomeRule &cr) :
	mainProgramEx(cr.getMainPR().totalLen), ADFEx(cr.getADFPR().size()) 
{
	initHelp(cr);
}

void Chromosome::init(ChromosomeRule &cr) {
	initVectorHelp(mainProgramEx, cr.getMainPR().totalLen);
	initVectorHelp(ADFEx, cr.getADFPR().size());

	initHelp(cr);
}

void Chromosome::initHelp(ChromosomeRule &cr) {
	//first init ADFPR`s element ex`s lenth
	vector<ProgramRule>& ADFPR = cr.getADFPR();
	for (int i = 0; i < ADFPR.size(); ++i) {
		vector<int>tmp(ADFPR[i].totalLen);
		ADFEx[i] = tmp;
	}
}
Chromosome& Chromosome::operator = (const Chromosome &c) {
	this->mainProgramEx = c.mainProgramEx;
	this->ADFEx = c.ADFEx;
	return *this;
}

File: pySource/gepChromosome/Chromosome.py
from gepChromosome import ChromosomeRule
import numpy as np
class Chromosome:
    def __init__(self,chromosomeRule = None):
        self.mainProgramEx = []
        self.ADFEx = []
        if chromosomeRule is not None:
            self.init(chromosomeRule)

    def init(self,chromosomeRule):
        self.mainProgramEx = np.zeros((chromosomeRule.getMainPR().totalLen),dtype=np.uint32)
        for i in range(0,len(chromosomeRule.getADFPR())):
            self.ADFEx.append(np.zeros(chromosomeRule.getADFPR()[i].totalLen,dtype=np.uint32))

